src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -84,6 +93,13 @@ private ModelMetadataCacheEntry CreateCacheEntry(ModelMetadataIdentity key)
             return new ModelMetadataCacheEntry(metadata, details);
         }
 
+        private ModelMetadata GetMetadataForObjectType()
+        {
+            var key = ModelMetadataIdentity.ForType(typeof(object));
+            var entry = CreateCacheEntry(key);
```

Comment:
GetMetadataForProperties() doesn't use the DefaultModelMetadata for the container's ModelType. It uses DefaultModelMetadataDetails that has less info that the Metadata. Wondering if we can reuse the Object Metadata if any of the properties are of type object?